Extensions/xinbox.js

```diff
@@ -1064,6 +1064,44 @@ XKit.extensions.xinbox = new Object({
 		document.body.appendChild(script);
 	},
 
+	do_post_colors: function(){
+		var colors = XKit.extensions.xinbox.get_blog_colors();
+		$(".posts .post[data-tumblelog]").not(":has(.xinbox-color-strip)").each(function(j,i){
+			var blog = i.dataset.tumblelog;
+
+			var n = '<div class="xinbox-color-strip" ' +
+				 'style="position: absolute; right: 0; top: 0; '+
+				 'border-top: 30px solid '+colors[blog]+'; border-left: 30px solid transparent;"'+
+				 'title="Submitted to '+blog+'"></div>';
+
+			$(i).append(n);
+		});
+	},
+
+	get_blog_colors: function() {
+		var current = XKit.storage.get("xinbox", "stable_blogs");
+		var new_blogs = XKit.tools.get_blogs();
+		if (!current) {
+			current = new_blogs;
+		} else {
+			current = current.split(";");
+			var set = new Set(current);
```

Comment:
@hobinjk 38 has been out since October, 2014. Is it really not safe to use?

On Mon, Aug 17, 2015 at 12:58 PM James Hobin <notifications@github.com>
wrote:

> In Extensions/xinbox.js
> <https://github.com/new-xkit/XKit/pull/304#discussion_r37216202>:
>
> > +				 'style="position: absolute; right: 0; top: 0; '+
> > +				 'border-top: 30px solid '+colors[blog]+'; border-left: 30px solid transparent;"'+
> > +				 'title="Submitted to '+blog+'"></div>';
> > +
> > +			$(i).append(n);
> > +		});
> > +	},
> > +
> > +	get_blog_colors: function() {
> > +		var current = XKit.storage.get("xinbox", "stable_blogs");
> > +		var new_blogs = XKit.tools.get_blogs();
> > +		if (!current) {
> > +			current = new_blogs;
> > +		} else {
> > +			current = current.split(";");
> > +			var set = new Set(current);
>
> Set()'s cross-browser compatibility seems pretty bad on Chrome (only
> available since chrome 38 when we technically support version since 27). A
> more standard approach like, current.forEach(function(c) { set[c] = true;
> }) seems preferable.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/new-xkit/XKit/pull/304/files#r37216202>.
>